File: Python/1ConcatSum.py
# Calculate the sum of every a[i] ° a[j] 
# concatenation for, string a = [10,2]

def sum(a, n):
    cnt = dict()
    ans = 0
    pre_sum = 0
    for i in range(n):
        ans += (i * a[i]) - pre_sum
        pre_sum += a[i]
 
        if (a[i] - 1) in cnt:
            ans -= cnt[a[i] - 1]

        if (a[i] + 1) in cnt:
            ans += cnt[a[i] + 1]
 
        if a[i] not in cnt:
            cnt[a[i]] = 0
        cnt[a[i]] += 1
     
    return ans
 
if __name__ == '__main__':
    a = [1, 2, 3, 1, 3]
    n = len(a)
    print(sum(a, n))

File: JavaScript/1ConcatSum.js
// Calculate the sum of every a[i] ° a[j] 
// concatenation for, string a = [10,2]

let arr = [10,2]
let n = arr.length
let sum = 0

function concatSum(arr){
for(i=0;i<n;i++)
      for(j=0;j<n;j++)
           sum += Number(''+arr[j]+arr[i]);
           return sum
      }

console.log(concatSum(arr))